### DISCUSSION TITLE
Extension Compatibility

### DISCUSSION THREAD

**Discussion by Leon0402** [eclipse-theia/theia#8601] [Q&A] [answered]
Hi, 

I tried to install an extension with theia (https://open-vsx.org/extension/llvm-vs-code-extensions/vscode-clangd) and got the error: "The language client requires VS Code version ^1.46.0 but received version 1.44.0"

So it seems like this plugin is built against a specific version of Vs Code. Now I wondered how theia works with this? When will this version be bumped?

I use theia 1.6 as an electron app.

**Comment by vince-fugnitto** [comment, marked as answer]
@Leon0402 the framework defines a default API version which is supported:

https://github.com/eclipse-theia/theia/blob/56c158dc7f87e6e9c33b194512969d2d5a78de42/packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts#L22

There is ongoing work to fetch extension versions which satisfy this API version, else you can update your application to bump the version, or start the application with a specific version through the `cli`:

https://github.com/eclipse-theia/theia/blob/56c158dc7f87e6e9c33b194512969d2d5a78de42/packages/plugin-ext-vscode/src/node/plugin-vscode-cli-contribution.ts#L32-L39

The version should be bumped upstream when we are confident the 'majority' of the VS Code API is supported for the given version.

**Reply by Leon0402** [reply]
Hmm I wonder though if then extensions, which require higher versions shouldn't be installed? 
I would expect that I can install older versions of extensions, which are compatible. 

How it the compatibilty tracked? Is there some kind of list showing missing stuff (which would also enable developers to contribute)?

**Reply by vince-fugnitto** [reply]
As I mentioned, it's currently in progress to only install versions which are API-compatible when searching for them in the `Extensions` view.
- https://github.com/eclipse-theia/theia/issues/7464
- https://github.com/eclipse-theia/theia/pull/8191

You may look at issues labeled [vscode](https://github.com/eclipse-theia/theia/issues?q=is%3Aissue+is%3Aopen+label%3Avscode) which are generally used to refer to issues regarding API compatibility. 

I believe the following document will also illustrate missing APIs: https://che-incubator.github.io/vscode-theia-comparator/status.html
